lib/Onyx.d.ts

```diff
@@ -0,0 +1,278 @@
+ * Storage.setItem() from Onyx.clear() will have already finished and the merged
+ * value will be saved to storage after the default value.
+ *
+ * @param keysToPreserve is a list of ONYXKEYS that should not be cleared with the rest of the data
+ */
+declare function clear(keysToPreserve?: OnyxKey[]): Promise<void>;
+
+/**
+ * Merges a collection based on their keys
+ *
+ * @example
+ *
+ * Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT, {
+ *     [`${ONYXKEYS.COLLECTION.REPORT}1`]: report1,
+ *     [`${ONYXKEYS.COLLECTION.REPORT}2`]: report2,
+ * });
+ *
+ * @param collectionKey e.g. `ONYXKEYS.COLLECTION.REPORT`
+ * @param collection Object collection keyed by individual collection member keys and values
+ */
+declare function mergeCollection<TKey extends CollectionKey, TMap>(collectionKey: TKey, collection: Collection<TKey, TMap, PartialDeep<KeyValueMapping[TKey]>>): Promise<void>;
+
+/**
+ * Insert API responses and lifecycle data into Onyx
+ *
+ * @param data An array of objects with shape {onyxMethod: oneOf('set', 'merge', 'mergeCollection'), key: string, value: *}
+ * @returns resolves when all operations are complete
+ */
+declare function update<TKeyList extends OnyxKey[]>(data: OnyxUpdates<[...TKeyList]>): Promise<void>;
+
+/**
+ * Initialize the store with actions and listening for storage events
+ *
+ * @param [options={}] config object
+ * @param [options.keys={}] `ONYXKEYS` constants object
+ * @param [options.initialKeyStates={}] initial data to set when `init()` and `clear()` is called
+ * @param [options.safeEvictionKeys=[]] This is an array of keys
+ * (individual or collection patterns) that when provided to Onyx are flagged
+ * as "safe" for removal. Any components subscribing to these keys must also
+ * implement a canEvict option. See the README for more info.
+ * @param [options.maxCachedKeysCount=55] Sets how many recent keys should we try to keep in cache
+ * Setting this to 0 would practically mean no cache
+ * We try to free cache when we connect to a safe eviction key
+ * @param [options.captureMetrics] Enables Onyx benchmarking and exposes the get/print/reset functions
+ * @param [options.shouldSyncMultipleInstances] Auto synchronize storage events between multiple instances
+ * of Onyx running in different tabs/windows. Defaults to true for platforms that support local storage (web/desktop)
+ * @param [options.debugSetState] Enables debugging setState() calls to connected components.
+ * @example
+ * Onyx.init({
+ *     keys: ONYXKEYS,
+ *     initialKeyStates: {
+ *         [ONYXKEYS.SESSION]: {loading: false},
+ *     },
+ * });
+ */
+declare function init(config?: InitOptions): void;
+
+declare const Onyx: {
+    connect: typeof connect;
+    disconnect: typeof disconnect;
+    set: typeof set;
+    multiSet: typeof multiSet;
+    merge: typeof merge;
+    mergeCollection: typeof mergeCollection;
+    update: typeof update;
+    clear: typeof clear;
+    getAllKeys: typeof getAllKeys;
```

Comment:
I see that we haven't provided declarations for methods which are only used by Onyx internally, so we declare the type for getAllKeys, and some of these other functions?

Reply:
Oh nevermind, I missed the declaration of Onyx above https://github.com/Expensify/react-native-onyx/blob/8dba043803a6269e90148519a9bc26838e7dda1e/lib/Onyx.js#L1291-L1307